exonum-java-binding/testkit/src/test/java/com.exonum.binding.testkit/TestKitTest.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright 2019 The Exonum Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exonum.binding.testkit;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import com.exonum.binding.service.AbstractServiceModule;
+import com.exonum.binding.service.Node;
+import com.exonum.binding.service.Service;
+import com.exonum.binding.service.ServiceModule;
+import com.exonum.binding.service.TransactionConverter;
+import com.exonum.binding.transaction.RawTransaction;
+import com.exonum.binding.transaction.Transaction;
+import com.google.inject.Singleton;
+import io.vertx.ext.web.Router;
+import org.junit.jupiter.api.Test;
+import java.util.ArrayList;
+import java.util.List;
+
+class TestKitTest {
+
+  @Test
+  void createTestKitForSingleService() {
+    TestKit testKit = TestKit.forService(TestServiceModule.class);
+    Service service = testKit.getService(TestService.SERVICE_ID, TestService.class);
+    assertEquals(service.getId(), TestService.SERVICE_ID);
+    assertEquals(service.getName(), TestService.SERVICE_NAME);
+  }
+
+  @Test
+  void createTestKitWithBuilderForSingleService() {
+    TestKit testKit = TestKit.builder(EmulatedNodeType.VALIDATOR)
+        .withService(TestServiceModule.class)
+        .build();
+    Service service = testKit.getService(TestService.SERVICE_ID, TestService.class);
+    assertEquals(service.getId(), TestService.SERVICE_ID);
+    assertEquals(service.getName(), TestService.SERVICE_NAME);
+  }
+
+  @Test
+  void createTestKitWithBuilderForMultipleSameServices() {
+    List<Class<? extends ServiceModule>> serviceModules = new ArrayList<>();
+    serviceModules.add(TestServiceModule.class);
+    serviceModules.add(TestServiceModule.class);
```

Comment:
Shall we allow that? With the present framework version it will not work at all.

Reply:
IIRC there was a discussion on wiki some time ago stating that we should, but that could change. What's the current outlook on having several instances of the same service?

Reply:
Even with multiple services, it is not possible to create two instances of the same service because their database schemas will be the same; their ids will be the same (so the testkit will likely fail somewhere in native). Dynamic services will require providing _instance_ parameters — i.e., something on top of a ServiceModule (e.g., a service instance name, or other initialization parameters), therefore the present interface won't work with them.